### ISSUE TITLE
Deprecate low value rules

### ISSUE THREAD

**Issue by quentin-jaquier-sonarsource** [SonarSource/SonarJS#1810]
After a first round of deprecation (#1698), we realized that some additional rules are behaving poorly and don't provide a lot of value.

- [S3002](https://jira.sonarsource.com/browse/RSPEC-3002): _Unary operators "+" and "-" should not be used with objects_

This rule is hardly ever reporting any issue, and when it does, it's questionable since it can be correct to use unary operator on object, for example:
```
let x = {toString() {return "42";}};
+x;
```
does not result in NaN.

- [S2583](https://jira.sonarsource.com/browse/RSPEC-2583): _Conditionally executed blocks should be reachable_

In [S2589](https://jira.sonarsource.com/browse/RSPEC-2589), we already report when a condition is always true or false, if it results in dead code, it's superfluous to report it a second time.
We therefore decided to deprecate this rule, and keep only [S2583](https://jira.sonarsource.com/browse/RSPEC-2589).

- [S2873](https://jira.sonarsource.com/browse/RSPEC-2873): _Calls should not be made to non-callable values_

We realized that this rule was never behaving very well, most of the issues that we could observe were false positive. Even with the current migration improvement, we will not have the tools to improve the results to a state where it's worth the migration effort, we therefore decided to deprecate it.

-  [S2714](https://jira.sonarsource.com/browse/RSPEC-2714): _Element type selectors should not be used with class selectors_
This rule is related to JQuery, we missed it during the first path of deprecation. See #1698.

## Security Rules
- [S2611](https://jira.sonarsource.com/browse/RSPEC-2611), Untrusted content should not be included, see #1859
 
- [S3271](https://jira.sonarsource.com/browse/RSPEC-3271), see #1862
